### ISSUE TITLE
diag_l0_*_send() ; assert if !len

### ISSUE THREAD

**Issue by fenugrec** [fenugrec/freediag#68]
all (?) L0 drivers have something like

```
	if (len <= 0) {
		return diag_iseterr(DIAG_ERR_BADLEN);
	}
```

but len is unsigned (size_t).
Anyway that should be an assert(0) unrecoverable error; L0 should never get 0-length sends.

Same idea in tty_read and tty_write

**Comment by fenugrec** [comment]
Once fixed, remove cppcheck suppression (see #70 )

**Comment by fenugrec** [comment]
Fixed via #70 .
